controllers/metal3.io/preprovisioningimage_controller.go

```diff
@@ -116,12 +116,26 @@ func (r *PreprovisioningImageReconciler) update(img *metal3.PreprovisioningImage
 	return false, err
 }
 
-func getImageURL() (url string, format metal3.ImageFormat, errorMessage string) {
-	format = metal3.ImageFormatISO
-	if iso := os.Getenv("DEPLOY_ISO_URL"); iso != "" {
-		url = iso
-	} else {
-		errorMessage = "No DEPLOY_ISO_URL specified"
+func getImageURL(acceptFormats []metal3.ImageFormat) (url string, format metal3.ImageFormat, errorMessage string) {
+	for _, fmt := range acceptFormats {
+		switch fmt {
+		case metal3.ImageFormatISO:
+			if iso := os.Getenv("DEPLOY_ISO_URL"); iso != "" {
+				return iso, fmt, ""
+			}
+			if errorMessage == "" {
+				format = fmt
+				errorMessage = "No DEPLOY_ISO_URL specified"
+			}
+		case metal3.ImageFormatInitRD:
+			if initrd := os.Getenv("DEPLOY_RAMDISK_URL"); initrd != "" {
+				return initrd, fmt, ""
+			}
+			if errorMessage == "" {
+				format = fmt
+				errorMessage = "No DEPLOY_RAMDISK_URL specified"
+			}
```

Comment:
If there are two kinds of `acceptformats`, and both of them fail to obtain the URL, Is it bettert to return more detailed error information (e.g Neither DEPLOY_ISO_URL and DEPLOY_RAMDISK_URL specified ) ?  This is convenient for users to better understand the current situation.

Reply:
This controller won't start at all unless one of these is provided, so the case you mention can't happen:
https://github.com/metal3-io/baremetal-operator/pull/1019/files#diff-1d7da700aa97796ea2261b0c8bedf79a7eeb24ee0dd2fff6782b9b2811c9455dR238-R247

Reply:
Fine, lgtm